library/cmac.c

```diff
@@ -468,6 +470,8 @@ int mbedtls_aes_cmac_prf_128( const unsigned char *key, size_t key_length,
 }
 #endif /* MBEDTLS_AES_C */
 
+#endif /* !MBEDTLS_CMAC_ALT */
```

Comment:
Note that `cmac_generate_subkeys()` is also used in  the [self_test](https://github.com/ARMmbed/mbedtls/pull/866/files#diff-97f39c1c46fcba3c19208c069f04a40fR809). You should consider taking this function out of `MBEDTLS_CMAC_ALT` check.
I would consider putting this function within:
~~~
#if ! defined(MBEDTLS_CMAC_ALT) || defined(MBEDTLS_SELF_TEST)
~~~